go/runtime/txpool/txpool.go

```diff
@@ -0,0 +1,808 @@
+	batch := t.checkTxQueue.GetBatch()
+	if len(batch) == 0 {
+		return
+	}
+
+	bi, err := t.getCurrentBlockInfo()
+	if err != nil {
+		t.logger.Warn("failed to get current block info, unable to check transactions",
+			"err", err,
+		)
+		return
+	}
+
+	rawTxBatch := make([][]byte, 0, len(batch))
+	for _, item := range batch {
+		rawTxBatch = append(rawTxBatch, item.Tx)
+	}
+	results, err := rr.CheckTx(ctx, bi.RuntimeBlock, bi.ConsensusBlock, bi.Epoch, bi.ActiveDescriptor.Executor.MaxMessages, rawTxBatch)
+	if err != nil {
+		t.logger.Warn("transaction batch check failed",
+			"err", err,
+		)
+		// NOTE: We do not send the results back as the batch will be retried.
+		return
+	}
+
+	// Remove the checked transaction batch.
+	t.checkTxQueue.RemoveBatch(batch)
+	pendingCheckSize.With(t.getMetricLabels()).Set(float64(t.PendingCheckSize()))
+
+	txs := make([]*transaction.CheckedTransaction, 0, len(results))
+	isLocal := make([]bool, 0, len(results))
+	var unschedule []hash.Hash
+	for i, res := range results {
+		// Send back the result of running the checks.
+		if batch[i].NotifyCh != nil {
+			batch[i].NotifyCh <- &results[i]
+			close(batch[i].NotifyCh)
+		}
+
+		if !res.IsSuccess() {
+			t.logger.Debug("check tx failed",
+				"tx", batch[i].Tx,
+				"result", res,
+				"recheck", batch[i].Meta.Recheck,
+			)
+
+			// If this was a recheck, make sure to remove the transaction from the scheduling queue.
+			if batch[i].Meta.Recheck {
+				unschedule = append(unschedule, batch[i].TxHash)
+			}
+			continue
+		}
+
+		if batch[i].Meta.Discard || batch[i].Meta.Recheck {
+			continue
+		}
+
+		txs = append(txs, res.ToCheckedTransaction(rawTxBatch[i]))
+		isLocal = append(isLocal, batch[i].Meta.Local)
+	}
+
+	// Unschedule any transactions that are being rechecked and have failed checks.
+	t.RemoveTxBatch(unschedule)
+
+	if len(txs) == 0 {
+		return
+	}
+
+	t.logger.Debug("checked new transactions",
+		"num_txs", len(txs),
+	)
+
+	// Queue checked transactions for scheduling.
+	for i, tx := range txs {
+		t.schedulerLock.Lock()
+		// NOTE: Scheduler exists as otherwise there would be no current block info above.
+		if err := t.scheduler.QueueTx(tx); err != nil {
+			t.schedulerLock.Unlock()
+			t.logger.Error("unable to schedule transaction", "tx", tx)
+			continue
+		}
+		t.schedulerLock.Unlock()
+
+		// Publish local transactions immediately.
+		publishTime := time.Now()
+		if isLocal[i] {
+			if err := t.txPublisher.PublishTx(ctx, tx.Raw()); err != nil {
+				t.logger.Warn("failed to publish local transaction",
+					"err", err,
+					"tx", tx,
+				)
+
+				// Sine publication failed, make sure we retry early.
+				t.republishCh.In() <- struct{}{}
+				publishTime = time.Time{}
+			}
+		}
+
+		if err := t.seenCache.Put(tx.Hash(), publishTime); err != nil {
+			// cache.Put can only error if capacity in bytes is used and the
+			// inserted value is too large. This should never happen in here.
```

Comment:
Yeah this error just needs to be ignored (as it can actually never happen because of the LRU cache configuration). Updated the comment and removed logging.